specification/containerservice/resource-manager/Microsoft.ContainerService/stable/2021-05-01/managedClusters.json

```diff
@@ -2346,71 +2079,67 @@
         },
         "podSubnetID": {
-          "$ref": "#/definitions/ContainerServiceVnetSubnetID",
-          "description": "Pod SubnetID specifies the VNet's subnet identifier for pods."
+          "type": "string",
+          "title": "The ID of the subnet which pods will join when launched.",
+          "description": "If omitted, pod IPs are statically assigned on the node subnet (see vnetSubnetID for more details). This is of the form: /subscriptions/{subscriptionId}/resourceGroups/{resourceGroupName}/providers/Microsoft.Network/virtualNetworks/{virtualNetworkName}/subnets/{subnetName}"
         },
         "maxPods": {
           "type": "integer",
           "format": "int32",
-          "description": "Maximum number of pods that can run on a node."
+          "description": "The maximum number of pods that can run on a node."
         },
         "osType": {
-          "$ref": "#/definitions/OSType",
-          "description": "OsType to be used to specify os type. Choose from Linux and Windows. Default to Linux."
+          "$ref": "#/definitions/OSType"
         },
         "osSKU": {
-          "$ref": "#/definitions/OSSKU",
-          "description": "OsSKU to be used to specify os sku. Choose from Ubuntu(default) and CBLMariner for Linux OSType. Not applicable to Windows OSType."
+          "$ref": "#/definitions/OSSKU"
         },
         "maxCount": {
           "type": "integer",
           "format": "int32",
-          "description": "Maximum number of nodes for auto-scaling"
+          "description": "The maximum number of nodes for auto-scaling"
         },
         "minCount": {
           "type": "integer",
           "format": "int32",
-          "description": "Minimum number of nodes for auto-scaling"
+          "description": "The minimum number of nodes for auto-scaling"
         },
         "enableAutoScaling": {
           "type": "boolean",
           "description": "Whether to enable auto-scaler"
         },
         "type": {
-          "$ref": "#/definitions/AgentPoolType",
-          "description": "AgentPoolType represents types of an agent pool"
+          "$ref": "#/definitions/AgentPoolType"
         },
         "mode": {
-          "$ref": "#/definitions/AgentPoolMode",
-          "description": "AgentPoolMode represents mode of an agent pool"
+          "$ref": "#/definitions/AgentPoolMode"
         },
         "orchestratorVersion": {
           "type": "string",
-          "description": "Version of orchestrator specified when creating the managed cluster."
+          "description": "The version of the orchestrator specified when creating the managed cluster."
```

Comment:
Is this field always `readOnly`? If not... what does it mean? what do I set it do? Why do I care?

Reply:
no you can mutage this to upgrade the cluster. its the k8s version of the closter. Since we only support k8s now we should just say k8s instead of orchestrator.

Reply:
Found good docs for this and the ManagedCluster `kubernetesVersion` field as well.